robot-agent/resourceMonitor.js

```diff
@@ -116,6 +162,37 @@ class ResourceMonitor {
     }, SAMPLE_RATE);
   }
 
+  startMonitoringFromPackage(packageName) {
+    if (this.monitoredPackages[packageName]) {
+      log.warn(`Already monitoring package ${packageName}, skipping`);
+      return;
+    }
+
+    // Use pgrep to find the PID of the running package
+    const { spawn } = require('child_process');
+    const pgrep = spawn('pgrep', ['-nf', `startPackage.sh ${packageName}`, '-U', process.getuid()]);
+    
+    let packagePid = null;
+    
+    pgrep.stdout.on('data', (data) => {
+      packagePid = parseInt(data.toString().trim());
+      log.debug(`Found running package ${packageName} with PID: ${packagePid}`);
+    });
+
+    pgrep.on('exit', (code) => {
+      if (code === 0 && packagePid) {
+        // Package is running, start monitoring
+        this.startMonitoring(packageName, packagePid);
+      } else {
+        log.debug(`Package ${packageName} not found running (exit code: ${code})`);
+      }
+    });
+
+    pgrep.on('error', (err) => {
+      log.error(`Error searching for package ${packageName}:`, err);
+    });
+  }
```

Comment:
This seems to replicate logic we already have in mqtt.js for reliably getting the PID if the capability is running.